pkg/querier/blocks_store_queryable.go

```diff
@@ -749,6 +752,8 @@ func (q *blocksStoreQuerier) fetchSeriesFromStores(ctx context.Context, sp *stor
 		streams       []storegatewaypb.StoreGateway_SeriesClient
 	)
 
+	debugContinuousTest := queryLimiter.DebugContinuousTest()
```

Comment:
Having this on the `QueryLimiter` seems out of place.

What if we had a "enable debug mode" flag that was added to the context instead?